### DISCUSSION TITLE
[BUG] MSBuild.exe (Visual Studio 2019) on Windows-latest is not in system PATH

### DISCUSSION THREAD

**Discussion by ldorau** [community/community#25540] [Actions]
<p>BUG: MSBuild.exe (Visual Studio 2019) on Windows-latest is not in system PATH</p>
<p>&gt; msbuild src$Env:solutionname -property:Configuration=Debug -m -v:m<br>
msbuild : The term ‘msbuild’ is not recognized as the name of a cmdlet, function, script file, or operable program.<br>
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.<br>
At D:\a_temp\9d00f860-8402-433d-a27e-c4564718cff2.ps1:2 char:1</p>
<ul>
<li>msbuild src$Env:solutionname -property:Configuration=Debug -m -v:m</li>
<li>
<pre><code class="lang-auto"></code></pre>
</li>
<li>CategoryInfo : ObjectNotFound: (msbuild:String) [], ParentContainsErrorRecordException</li>
<li>FullyQualifiedErrorId : CommandNotFoundException<br>
##[error]Process completed with exit code 1.</li>
</ul>
<p><a href="https://github.com/ldorau/pmdk/commit/ea64967772ab6a6f2024ba26d8e48ea7893d93b7/checks?check_suite_id=339247591" rel="nofollow noopener">https://github.com/ldorau/pmdk/commit/ea64967772ab6a6f2024ba26d8e48ea7893d93b7/checks?check_suite_id=339247591</a></p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Thank you for being here! This error is due to msbuild is not added to the path correctly. On Github runner, msbuild is not added into PATH by default.</p>
<p>Add solution for other guys:</p>
<ol>
<li>
<p>add the env path, code as below:</p>
<p>echo “::add-path::C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin”</p>
</li>
<li>
<p>or use action ‘warrenbuckley/Setup-MSBuild’ before msbuild command.</p>
</li>
</ol>

**Comment by chrispat** [comment]
<p>Visual Studio does not put MSBuild in the path so I don’t think it is appropriate that we do. You should reference MSBuild from its fully qualified location in order to make sure you are running the right version.</p>
